crates/cxx-qt-lib/src/gui/qpainter.rs

```diff
@@ -0,0 +1,187 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#[cxx::bridge]
+mod ffi {
+    #[namespace = "Qt"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qt.h");
+        type LayoutDirection = crate::LayoutDirection;
+        type BGMode = crate::BGMode;
+        type ClipOperation = crate::ClipOperation;
+    }
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qpainter.h");
+        type QPainter;
+        include!("cxx-qt-lib/qrect.h");
+        type QRect = crate::QRect;
+        include!("cxx-qt-lib/qrectf.h");
+        type QRectF = crate::QRectF;
+        include!("cxx-qt-lib/qpoint.h");
+        type QPoint = crate::QPoint;
+        include!("cxx-qt-lib/qline.h");
+        type QLine = crate::QLine;
+        include!("cxx-qt-lib/qcolor.h");
+        type QColor = crate::QColor;
+        include!("cxx-qt-lib/qimage.h");
+        type QImage = crate::QImage;
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+
+        /// Returns the current background mode.
+        #[rust_name = "background_mode"]
+        fn backgroundMode(self: &QPainter) -> BGMode;
+
+        /// Returns the currently set brush origin.
+        #[rust_name = "brush_origin"]
+        fn brushOrigin(self: &QPainter) -> QPoint;
+
+        /// Returns the bounding rectangle of the current clip if there is a clip;
+        /// otherwise returns an empty rectangle. Note that the clip region is given in logical coordinates.
+        #[rust_name = "clip_bounding_rect"]
+        fn clipBoundingRect(self: &QPainter) -> QRectF;
```

Comment:
In cases where the Qt API returns an uninitialized or otherwise "empty" object if something is not set, we usually opt to wrap the function for a more Rust-y API (i.e. returning `Option`).

This would make sense here as well. So please use: `clip_bound_rect_or_empty` as the Rust name and create a small wrapper function:
```rust
impl QPainter {
    fn clip_bounding_rect(&self) -> Option<QRectF> {
        let result = self.clip_bounding_rect_or_empty();
        if result.is_valid() {
            Some(result)
        } else {
            None
        }
    }
}
```